### PR TITLE
Get type information from dnanexus-executable.json if possible

### PR DESCRIPTION
**Pull request by jdidion**

### CODE REVIEW

File: src/main/scala/dx/core/ir/TypeSerde.scala [hunk 1]
```diff
@@ -219,6 +219,57 @@ object TypeSerde {
     }
   }
 
+  private def fromNativeNonArray(cls: String): Type = {
```
[the remark this answers is not in the document]
Comment (jdidion) [reply src/main/scala/dx/core/ir/TypeSerde.scala:222]: Directory is not a native DNAnexus type. We still haven't figured out exactly how we're going to handle directories. I added another PR with an updated README that mentions this limitation.

**Review by commandlinegirl — DISMISSED** [review]
LGTM, thanks for this optimization!

**Review by commandlinegirl — APPROVED** [review]

**Review by commandlinegirl — CHANGES_REQUESTED** [review]
There are two tests that are failing:

image

I'll need to check why the test suite didn't fail..

**Review by commandlinegirl — APPROVED** [review]
Thank you for the fix!

**Pull request merged**